### DISCUSSION TITLE
How do you reset the state machine?

### DISCUSSION THREAD

**Discussion by owenashurst** [statelyai/xstate#2723] [Q&A]
How does one reset the state machine so that it's rendering the initial view? I can't seem to find any function to do this. Seems like the only way is to put "home" on every on: {} object for every state which just seems messy.

**Comment by Andarist** [comment]
If you are using React then you could leverage `key` prop on your component that owns the machine.

The idiomatic XState way to do that would be to create smth like:
```js
createMachine({
  initial: 'start',
  states: {
    start: {
      entry: assign(YOUR_INITIAL_CONTEXT)
    }
  },
  on: {
    RESET: 'start'
  }
})
```

**Reply by davidkpiano** [reply]
Also in React, you can "recreate" the component, since this seems to be the intention. You do that by giving the component a new `key={...}` prop.
